src/controllers/dlgprefcontroller.cpp

```diff
@@ -816,20 +828,32 @@ void DlgPrefController::slotShowMapping(std::shared_ptr<LegacyControllerMapping>
     m_pInputProxyModel = pInputProxyModel;
     delete m_pInputTableModel;
     m_pInputTableModel = pInputModel;
+    connect(m_ui.inputControlSearch,
+            &QLineEdit::returnPressed,
+            this,
+            &DlgPrefController::slotInputControlSearch);
+    connect(m_ui.inputControlSearchBtn,
+            &QPushButton::clicked,
+            this,
+            &DlgPrefController::slotInputControlSearch);
```

Comment:
These connections can be done in the constructor. 
This is also an issue in https://github.com/ronso0/mixxx/tree/mapping-table-search_wsearch
after a while these controls are 10 times connected and a search is done that often slowing down the process. 
After moving them to the constructor there is no longer an extra GUI lag. 

Another possible issue is that `ControllerMappingTableProxyModel::filterAcceptsRow()` is quite slow. 
In case of searching during typing, it is called often but is acting always on the same data. 
Maybe there is a way to cache the heavy work of retreaving and fromating all strinsg over and over again.

Reply:
oh, true, I moved the conection to the constructor.

However that is not the issue with WSearchLineEdit since the GUI lockup occurs when the mapping is shown for the first time (only one conection) and no search signal has been fired, yet.
I guess it's related to some show/resize event or a conflicting QWidget/WWidget method, because occasionally I also experience another issue with the WSearchLineEdit in DlgDeveloperTools (dialog is extremly wide, ~10x screen width).
Since the current simple search works well I'm not motivated to debug that further right now.

Reply:
> In case of searching during typing, it is called often but is acting always on the same data.
Maybe there is a way to cache the heavy work of retreaving and fromating all strinsg over and over again.

With the simple search there is no search-as-you-type, it's only triggered by Return or the Search button (and after editing items).
Also, `data` and the delegates' `displayText` get called every time the view changes (scroll, show, hide ...) so I'm not sure if custom caching is worth the effort for this view of rather small data models (after all, even with huge mappings, each table holds like ~200 items max?).